File: java/src/com/mogreet/mercury/transaction/Send.java
/**
* Copyright 2011 Nicolas Payot
*
* This file is part of Mercury.
*
* Mercury is free software: you can redistribute it and/or modify
* it under the terms of the GNU General Public License as published by
* the Free Software Foundation, either version 3 of the License, or
* (at your option) any later version.
*
* Mercury is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
* GNU General Public License for more details.
*
* You should have received a copy of the GNU General Public License
* along with Mercury.  If not, see <http://www.gnu.org/licenses/>.
*/

package com.mogreet.mercury.transaction;

import com.mogreet.mercury.*;
import org.w3c.dom.*;
import javax.xml.xpath.*;
import java.lang.Math.*;

/**
* The Send object contains the response from a {@link Mercury#send} request to the Moms API.
*/
public class Send extends Response {
	
	private int messageId;
	private String hash;
	
	/**
	* Creates a new Send response object.
	* @param xmlDoc XML DOM document that is returned by all the requests.
	*/
	public Send(Document xmlDoc) {
		super(xmlDoc);
		if (!super.responseIsValid())
			throw new RuntimeException(super.getMessage());
		try {
			XPath xpath = XPathFactory.newInstance().newXPath();
			XPathExpression expr;

			expr = xpath.compile("/response/message_id/text()");
			this.messageId = ((Double)(expr.evaluate(xmlDoc, XPathConstants.NUMBER))).intValue();

			expr = xpath.compile("/response/hash/text()");
			this.hash = (String)(expr.evaluate(xmlDoc, XPathConstants.STRING));

		} catch (XPathExpressionException e) {
			throw new RuntimeException("\nAn error occured while parsing the XML data for the SEND call: " + e.getMessage());
		}
	}
	
	/**
	* @return The message ID.
	*/
	public int getMessageId() {
		return messageId;
	}
	
	/**
	* @return The hash.
	*/
	public String getHash() {
		return hash;
	}
}

File: dotNet/MogreetSDK/MogreetSDK/transaction/Send.cs
using System;
using System.Xml;
using System.Security;
using System.Collections.Generic;

namespace MogreetSDK.transaction
{
    /// <summary>
    /// The Send object contains the response from a <see cref="MogreetSDK.Mercury.send"/> request to the Moms API.
    /// </summary>
    /// <exception cref='Exception'>
    /// Represents errors that occur during application execution.
    /// </exception>
    public class Send : Response
    {
        private int messageId;
        private string hash;

        /// <summary>
        /// Initializes a new instance of the <see cref="SDK.Send"/> class.
        /// </summary>
        /// <param name='xmlDoc'>
        /// Xml document that is returned by all the requests.
        /// </param>
        /// <exception cref='Exception'>
        /// Represents errors that occur during application execution.
        /// </exception>
        public Send(XmlDocument xmlDoc)
            : base(xmlDoc)
        {
            if (!(base.responseIsValid()))
                throw new Exception(base.getMessage());
            try
            {
                // Creates XML Reader
                XmlNode messr = mXmlDoc.SelectSingleNode("//response/message_id/text()");
                this.messageId = Int32.Parse(messr.InnerText);
                Console.WriteLine("messageId: " + this.messageId);

                XmlNode hashr = mXmlDoc.SelectSingleNode("//response/hash/text()");
                this.hash = hashr.Value;
                Console.WriteLine("hash: " + this.hash);

            }
            catch (XmlException e)
            {
                throw new Exception("\nAn error occured while parsing the XML data for the SEND call: " + e.Message);
            }

        }

        /// <summary>
        /// Gets the message identifier.
        /// </summary>
        /// <returns>
        /// The message identifier.
        /// </returns>
        public int getMessageId()
        {
            return this.messageId;
        }

        /// <summary>
        /// Gets the hash.
        /// </summary>
        /// <returns>
        /// The hash.
        /// </returns>
        public string getHash()
        {
            return this.hash;
        }
    }
}
